lib/engine/game/g_18_carolinas/step/track.rb

```diff
@@ -8,76 +8,75 @@ module G18Carolinas
+            return [] if entity.company? || !can_lay_tile?(entity) && !conversion_available?
 
-            @mode == :new_track ? LAY_ACTIONS : CONVERT_ACTIONS
+            conversion_available? ? ALL_ACTIONS : LAY_ACTIONS
           end
 
-          def setup
-            @mode = :new_track
-            super
-          end
-
-          def update_tile_lists(tile, old_tile)
-            @game.update_tile_lists!(tile, old_tile)
+          def conversion_available?
+            @game.phase.available?('5') && @round.num_laid_track.zero?
           end
 
-          def mode_enabled?
-            return false if @round.num_laid_track.positive?
-
-            @game.phase.available?('5')
+          def round_state
+            super.merge(
+              {
+                convert_mode: nil,
+              }
+            )
           end
 
-          def mode_text
-            @mode == :new_track ? 'Track Conversion Mode' : 'Tile Lay/Upgrade Mode'
-          end
-
-          def change_mode
-            return :new_track unless @game.phase.available?('5')
-            return :new_track if @round.num_laid_track.positive?
-
-            @mode = @mode == :new_track ? :convert_segment : :new_track
+          def setup
+            super
+            @round.convert_mode = nil
           end
 
-          def instructions
-            'Click revenue centers, again to cycle paths. Must be from city/offboard to city/offboard'
+          def update_tile_lists(tile, old_tile)
+            @game.update_tile_lists!(tile, old_tile)
           end
 
-          def total_str(active_routes)
-            raise GameError, 'No routes' if active_routes.empty?
-
-            _rev = @game.routes_revenue(active_routes) # force check
-            'Convert Segment'
+          def choice_name
+            'Switch to'
           end
 
-          def revenue_fail
-            'Invalid Segment'
+          def choices
+            {
+              conversion: 'Track Conversion Mode',
+            }
           end
 
-          def process_run_routes(action)
-            hexes = action.routes[0].connection_hexes.flatten.uniq.map { |h| @game.hex_by_id(h) }
-            hexes_to_flip = hexes.select { |h| h.tile.paths.any? { |p| p.track != :broad } }
-            raise GameError, 'No tiles with Southern Track submitted' if hexes_to_flip.empty?
-
-            hexes_to_flip.each { |h| @game.flip_tile!(h) }
+          def process_choose(_action)
+            @round.convert_mode = true
+            @log << 'Switching to Track Conversion Mode'
             pass!
           end
 
-          def conversion?
-            true
-          end
-
-          def available_hex(entity, hex)
-            return super if @mode == :new_track
-
-            @game.graph_for_entity(entity).reachable_hexes(entity)[hex]
+          def legal_tile_rotation?(entity, hex, tile)
+            return super unless @game.phase.available?('5')
```

Comment:
this looks copy and pasted, what did you change?

Reply:
This line: old_paths.all? { |path| new_paths.any? { |p| @game.path_subset?(path, p) } }

Reply:
I was hesitant to modify tracker to pull this out as an overrideable method, but I can if you think it best.

Reply:
i think when you do copy and pasting and have to override other places like track_match? it probably indicates you're doing something wrong

all of this stuff is really just because you're trying to skip tracks_match? you could make this all much cleaner by setting a new path on variable to ignore tracks_match?, then you wouldn't have to change path walk and also copy and paste this method

let's make this change

Reply:
Except I don't always want to skip tracks_match?, or I'm not understanding your suggestion.